src/lava/lib/dnf/inputs/rate_code_spike_gen/models.py

```diff
@@ -0,0 +1,180 @@
+
+    a_in: PyInPort = LavaPyType(PyInPort.VEC_DENSE, float)
+    s_out: PyOutPort = LavaPyType(PyOutPort.VEC_DENSE, bool)
+
+    ts_last_changed: int = 1
+
+    def _compute_distances(self, pattern: np.ndarray) -> np.ndarray:
+        """Converts pattern representing spike rates in Hz to
+        inter spike distances in timesteps.
+
+        Assumes a minute contains TIME_STEPS_PER_MINUTE timesteps.
+        Assumes all spike rate values less than MIN_SPIKE_RATE are negligible.
+
+        Parameters
+        ----------
+        pattern : numpy.ndarray
+            pattern representing spike rates in (Hz)
+
+        Returns
+        -------
+        distances : numpy.ndarray
+            inter spike distances in (timesteps)
+
+        """
+        # Represent infinite inter spike distances (i.e negligible spike rates)
+        # as 0
+        distances = np.where(pattern > self.min_spike_rate,
+                             np.rint(TIME_STEPS_PER_MINUTE / pattern).astype(
+                                 np.int), 0)
+
+        return distances
+
+    def _compute_first_spike_times(self, distances):
+        """Randomly picks an array of first spike time given an array of
+        inter spike distances.
+
+        Every first spike time must be less than the associated
+        inter spike distance.
+
+        Parameters
+        ----------
+        distances : numpy.ndarray
+            inter spike distances in (timesteps)
+
+        Returns
+        -------
+        first_spike_times : numpy.ndarray
+            first spike time for each "neuron"
+
+        """
+        rng = np.random.default_rng(
+            seed=None if self.seed[0] == -1 else self.seed[0])
+
+        first_spikes = np.zeros_like(distances)
+
+        # Find indices where distance is 0 (where the should never be spikes)
+        idx_zeros = distances == 0
+
+        # Trick to yield a distances array in the right format for rng.integers
+        distances[~idx_zeros] = distances[~idx_zeros] + 1
+
+        # For indices where there should be a first spike ...
+        # Pick a random number in [1, distance[
+        first_spikes[~idx_zeros] = rng.integers(
+            low=np.ones_like(distances[~idx_zeros]), high=distances[~idx_zeros])
+
+        return first_spikes
+
+    def _generate_spikes(self,
+                         time_step: int) -> np.ndarray:
+        """Generates an array of bool values where True represent a spike
+        and False represents no-spike.
+
+        Uses internal state such as inter spike distances, first spike times,
+        last spike times and time step where last pattern change happened to
+        derive whether each "neuron" should fire.
+
+        Parameters
+        ----------
+        time_step : int
+            current time step
+
+        Returns
+        -------
+        spikes : numpy.ndarray
+            spikes array
+
+        """
+        spikes = np.zeros(self.spikes.shape, np.bool)
```

Comment:
change to `bool` to avoid deprecation warning.
There are a few instances of it as well as `np.int`

Reply:
Resolved